File: java-example/src/test/java/org/onflow/examples/java/getExecutionData/GetExecutionDataAccessAPIConnectorTest.java
package org.onflow.examples.java.getExecutionData;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.onflow.flow.common.test.FlowEmulatorProjectTest;
import org.onflow.flow.common.test.FlowTestClient;
import org.onflow.flow.sdk.*;

import static org.junit.jupiter.api.Assertions.*;

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
public class GetExecutionDataAccessAPIConnectorTest {

    @FlowTestClient
    private FlowAccessApi accessAPI;

    private GetExecutionDataAccessAPIConnector connector;
    private FlowId blockId;

    @BeforeEach
    public void setup() {
        connector = new GetExecutionDataAccessAPIConnector(accessAPI);

        // Set a specific block ID
        FlowBlock block;
        FlowAccessApi.AccessApiCallResponse<FlowBlock> response = accessAPI.getLatestBlock(true);
        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            block = ((FlowAccessApi.AccessApiCallResponse.Success<FlowBlock>) response).getData();
        } else {
            throw new RuntimeException(((FlowAccessApi.AccessApiCallResponse.Error) response).getMessage(), ((FlowAccessApi.AccessApiCallResponse.Error) response).getThrowable());
        }
        blockId = block.getId();
    }

    @Test
    public void canFetchExecutionDataByBlockId() {
        FlowExecutionResult executionData = connector.getExecutionDataByBlockId(blockId);

        assertNotNull(executionData, "Execution data should not be null");

        assertFalse(executionData.getChunks().isEmpty(), "Execution data should contain chunks");

        for (int chunkNo = 0; chunkNo < executionData.getChunks().size(); chunkNo++) {
            FlowChunk chunk = executionData.getChunks().get(chunkNo);
            assertTrue(chunk.getNumberOfTransactions() > 0, "Chunk " + chunkNo + " should contain transactions");
        }
    }
}

File: kotlin-example/src/test/kotlin/org/onflow/examples/kotlin/getExecutionData/GetExecutionDataAccessAPIConnectorTest.kt
package org.onflow.examples.kotlin.getExecutionData

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.onflow.flow.common.test.FlowEmulatorProjectTest
import org.onflow.flow.common.test.FlowTestClient
import org.onflow.flow.sdk.*

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
class GetExecutionDataAccessAPIConnectorTest {
    @FlowTestClient
    lateinit var accessAPI: FlowAccessApi

    private lateinit var connector: GetExecutionDataAccessAPIConnector
    private lateinit var blockId: FlowId

    @BeforeEach
    fun setup() {
        connector = GetExecutionDataAccessAPIConnector(accessAPI)

        // Set a specific block ID
        val block = when (val response = accessAPI.getLatestBlock()) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }
        blockId = block.id
    }

    @Test
    fun `Can fetch execution data by block ID`() {
        val executionData: FlowExecutionResult = connector.getExecutionDataByBlockId(blockId)

        assertNotNull(executionData, "Execution data should not be null")

        assertTrue(executionData.chunks.isNotEmpty(), "Execution data should contain chunks")

        executionData.chunks.forEachIndexed { chunkNo, chunk ->
            assertTrue(chunk.numberOfTransactions > 0, "Chunk $chunkNo should contain transactions")
        }
    }
}
